### PR TITLE
add more handrails around docker invocations

### PR DESCRIPTION
**Pull request by mgoldey**
Observed issues:
- docker can fail on removal of init containers - resolved by invoking in detached mode, waiting for exit, and then removing after it's exited
- docker compose often doesn't know how to tear down previous volumes and containers left on failure - resolved by always pruning containers then volumes on launch

### CODE REVIEW

File: launch.py [hunk 1]
```diff
@@ -113,14 +115,15 @@ def load_builtin_nlprocessor_models():
...
-        volumes={builtin_models.name: {
-            "bind": "/models/transformers",
-            "mode": "rw"
-        }},
+        auto_remove=False,
+        detach=True,
+        volumes={builtin_models.name: {"bind": "/models/transformers", "mode": "rw"}},
     )
+    initcontainer.start()
+    initcontainer.wait()
```
Comment (cbrochtrup) [inline launch.py:125]: How long is waiting? Should we have log messages to alert users of what we're waiting for?

**Review by cbrochtrup — APPROVED** [review]
Thanks for the continual improvements.

File: launch.py (hunk 1 above)
Comment (mgoldey) [reply launch.py:125]: This is a blocking wait until the container exits
https://docker-py.readthedocs.io/en/stable/containers.html#docker.models.containers.Container.wait

Comment (cbrochtrup) [reply launch.py:130]: Have you notices this takes a long time? If it takes a long time, I think we should add a log message alerting users who're waiting for the container to exit.

Comment (mgoldey) [reply launch.py:130]: It takes as long as the file copies take, which is a few seconds. The really long steps are the docker pull for init containers and the discovery startup. I'm personally holding off on adding further logging until we capture and prune the logging of docker compose, which overwhelms everything else.

Comment (cbrochtrup) [reply launch.py:130]: If the logs will overrun, then no need to add them.

**Pull request merged**